### ISSUE TITLE
How to add common folder in all functions

### ISSUE THREAD

**Issue by infinitedreams9586** [UnitedIncome/serverless-python-requirements#265]
There should be a way to specify common folder to include in all functions.
I have utility folder which is used by all functions.

**Comment by infinitedreams9586** [comment]
yes,  I would like to include a common folder, I tried with vendor, but instead of copying complete folder as it is, it cooies files inside vendor folder. it messes up the import statement in py files.

**Comment by infinitedreams9586** [comment]
for ex. if I have python file where import is like,

from common.db import tables

and add   vendor: common   instead of copying common folder as it is, it copies files inside it without common folder, so import statement above fails.

**Comment by pydanny** [comment]
I've found due to how Python packages and Lambda environments work, you have to create installable packages to support your use case. That's what prompted my pre-alpha [configure-django](https://github.com/pydanny/configure-django) package.

**Comment by pydanny** [comment]
Another idea is to do this with your vendor library:

```
vendored-libraries/
--common/
----thing.py
----utils.py
```

Then you can do things like: `from common.thing import do_action`.

**Comment by infinitedreams9586** [comment]
> Another idea is to do this with your vendor library:
> 
> ```
> vendored-libraries/
> --common/
> ----thing.py
> ----utils.py
> ```
> Then you can do things like: `from common.thing import do_action`.

In that case, it will not work on local. As import should be from vendored-library.common.thing import action. 
Plugin copies the files instead of complete folder, that's the issue.

**Comment by infinitedreams9586** [comment]
> I've found due to how Python packages and Lambda environments work, you have to create installable packages to support your use case. That's what prompted my pre-alpha [configure-django](https://github.com/pydanny/configure-django) package.

Could you let me know how to create installable package and use it in yml file ? without publishing the package to public.

**Comment by dschep** [comment]
So does it not work to have `/vendor/common/` and then `import common`?

**Comment by dschep** [comment]
Also, bc of how serverless handles them.. symlinks should be a workable solution too:
```
mkdir lambda1
mkdir lambda2
mkdir common
cd lambda1
ln -s ../common
cd ../lambda2
ln -s ../common
cd .. #done!
```

**Comment by dschep** [comment]
right, you'd have to set `PYTHONPATH=vendor` to run locally.

**Comment by dschep** [comment]
:+1: Glad you got it working @infinitedreams9586!

I'll leave this open for now as a reminder to write up some better documentation about using the vendor option.
